opennms-provision/opennms-detectorclient-rpc/src/main/java/org/opennms/netmgt/provision/detector/client/rpc/DetectorRequestBuilderImpl.java

```diff
@@ -99,11 +98,8 @@ public DetectorRequestBuilder withClassName(String className) {
 
     @Override
     public DetectorRequestBuilder withServiceName(String serviceName) {
-        final ServiceDetector detector = client.getRegistry().getDetectorByServiceName(serviceName);
-        if (detector == null) {
-            throw new IllegalArgumentException("No detector found with service name '" + serviceName + "'.");
-        }
-        this.className = client.getRegistry().getDetectorClassNameFromServiceName(serviceName);
+        client.getRegistry().getDetectorClassNameFutureFromServiceName(serviceName)
```

Comment:
We can't do this asynchronously in this method - there is no guarantee that `whenComplete` will be called before this is used further.

Reply:
I see.